components/session-service/server/server.go

```diff
@@ -454,17 +455,19 @@ func (s *Server) refreshHandler(w http.ResponseWriter, r *http.Request) {
 	// maybeExchangeRefreshTokenForIDToken will return the still-valid idToken.
 
 	// Check if the session exists
-	// Note 2018/01/05 (sr): Couldn't find a better way.
-	if keys, err := sess.Keys(); err != nil { // nolint: vetshadow
+	if alive, err := sess.GetBool(aliveKey); err != nil {
+		s.log.Error("error retrieving session alive key")
 		httpError(w, http.StatusInternalServerError)
 		return
-	} else if len(keys) == 0 {
+	} else if !alive {
```

Comment:
maybe I've lost track but does this need to be an `else if`? There's only two states, right?

Reply:
not worth blocking on tho

Reply:
The if branch is `err! = nil`,  when the thing couldn't be retrieved

Reply:
hmm the var `alive` assignment is confusing me. So this reads `if err is not nil, do this, else if it's not alive, do this` and then the rest is if `alive`? No extra `else` required?

Reply:
Correct @blakestier.
Would it clearer if we were to remove the `else`? (That's valid because the previous block includes a `return`.) Then it is:
```
if err is not nil do this and return
if !alive do that and return
// continue with "alive" case
```